src/business-logic/services/panel-service-implementation.ts

```
@@ -56,25 +56,43 @@ export class PanelServiceImplementation implements PanelService {
+      throw new UnsupportedOperationException(`Can't ${verb} PanelConfiguration. No PanelLayoutConfiguration exist for ${panelLayoutConfigurationId}`)
     }
+  }
 
-    if (panelLayoutConfiguration.model !== panelConfigurationWithoutId.model || panelLayoutConfiguration.type !== panelConfigurationWithoutId.type) {
-      throw new UnsupportedOperationException(
-        'Can\'t create PanelConfiguration. Both PanelModel and PanelType needs to match on PanelConfiguration and PanelLayoutConfiguration.'
-        + `PanelConfiguration: Model ${panelConfigurationWithoutId.model}, Type: ${panelConfigurationWithoutId.type} - `
-        + `PanelLayoutConfiguration: Model: ${panelLayoutConfiguration.model}, Type: ${panelLayoutConfiguration.type}`
-      )
+  private throwUnsupportedOperationWithIncompatibleMessage(panelConfiguration: PanelConfiguration, panelLayoutConfiguration: PanelLayoutConfiguration, verb: string): void {
+    throw new UnsupportedOperationException(
+      `Can't ${verb} PanelConfiguration. Both PanelModel and PanelType needs to match on PanelConfiguration and PanelLayoutConfiguration.`
+      + `PanelConfiguration: Model ${panelConfiguration.model}, Type: ${panelConfiguration.type} - `
+      + `PanelLayoutConfiguration: Model: ${panelLayoutConfiguration.model}, Type: ${panelLayoutConfiguration.type}`
+    )
+  }
+
+  public async updatePanelConfiguration(panelConfiguration: PanelConfiguration): Promise<void> {
+    const panelLayoutConfiguration: PanelLayoutConfiguration = await this.fetchPanelLayoutConfiguration(panelConfiguration.panelLayoutConfigurationId, 'update')
+
+    if (panelConfiguration.model !== panelLayoutConfiguration.model || panelConfiguration.type !== panelLayoutConfiguration.type) {
+      this.throwUnsupportedOperationWithIncompatibleMessage(panelConfiguration, panelLayoutConfiguration, 'update')
     }
```

Suggestion: Instead of having the check wherever we want to throw the exception, the condition could be moved to a method (e.g. merged into `throwUnsupportedOperationWithIncompatibleMessage`), so this method could simply call something like 
```typescript
this.assertPanelAndPanelLayoutCompatibility(panelConfiguration, panelLayoutConfiguration, 'update').
```

---

I have introduced `assertPanelLayoutConfigurationAndPanelConfigurationCompatibility(...)`